corehq/ex-submodules/pillowtop/reindexer/change_providers/couch.py

```diff
@@ -59,11 +59,30 @@ def data_function(**view_kwargs):
 
         args_provider = MultiKeyViewArgsProvider(keys, include_docs=True, chunk_size=self.chunk_size)
 
-        for row in paginate_function(data_function, args_provider, event_handler=self.event_handler):
+        for row in self._get_paginated_iterable(
+                data_function,
+                args_provider,
+                self.event_handler,
+                self._format_resumable_key(resumable_key)):
             yield Change(
                 id=row['id'],
                 sequence_id=None,
                 document=row.get('doc'),
                 deleted=False,
                 document_store=None
             )
+
+    @staticmethod
+    def _get_paginated_iterable(data_function, args_provider, event_handler=None, resumable_key=None):
+        if resumable_key:
+            return ResumableFunctionIterator(
+                resumable_key,
+                data_function,
+                args_provider,
+                lambda x: x.id,
+                event_handler=event_handler)
```

Comment:
cool! that was easy

nit: paren should be wrapped to new line

Reply:
haha, i'm actually not 100% sure which style i prefer

Reply:
732c3b9, there are definitely times that i think it looks nicer with the paren not wrapped